pkg/tfvars/internal/cache/cache.go

```diff
@@ -232,5 +231,17 @@ func DownloadFile(baseURL string, dataType string) (string, error) {
 func DownloadImageFile(baseURL string) (string, error) {
 	logrus.Infof("Obtaining RHCOS image file from '%v'", baseURL)
 
-	return DownloadFile(baseURL, imageDataType)
+	var u urlWithIntegrity
+	parsedURL, err := url.ParseRequestURI(baseURL)
+	if err != nil {
+		return "", err
+	}
+	q := parsedURL.Query()
+	if uncompressedSHA256, ok := q["sha256"]; ok {
+		u.uncompressedSHA256 = uncompressedSHA256[0]
+		delete(q, "sha256")
```

Comment:
If the intention is to remove the query value from `parsedURL`, this is not going to have that effect.
https://play.golang.org/p/vISRRGvej-w

Also, I think it is preferred to use the `Del` function.
```suggestion
		q.Del("sha256")
```

Reply:
Also needed to re-set the `RawQuery`.  Both fixed.